### PR TITLE
Morecov

### PR DESCRIPTION
**Pull request by darioizzo**
Paying the iron price

### CODE REVIEW

**Comment by darioizzo** [comment]
Good to go for me

File: include/pagmo/detail/custom_comparisons.hpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ inline bool equal_to_f(T a, T b)
 }
 
 // equal_to_vf than compares vectors of floating point types considering nan==nan
-template <typename T>
+template <typename T, detail::enable_if_is_floating_point<T> = 0>
```
Comment (bluescarni) [inline include/pagmo/detail/custom_comparisons.hpp]: The ``enable_if`` stuff is typically used in template specialisations, if you really want to have a check here it should go as a static assert in the body of the class:

```static_assert(std::is_floating_point<T>::value, "This class can be used only with floating-point types.");```

Comment (darioizzo) [inline include/pagmo/detail/custom_comparisons.hpp]: is it not going to work like that? or its just a matter of style?

Comment (bluescarni) [inline include/pagmo/detail/custom_comparisons.hpp]: It's not gonna do anything different from the static assertion, and it is more opaque. If this was a template specialisation, you could use it to pick another specialisation after you have disabled the invalid one, but in this case there's only one implementation.

**Pull request merged**